Python/CMakeLists.txt

```diff
@@ -737,59 +728,53 @@ set(PYTHON_HEADERS
-            ${PYTHON_SOURCES}       ${PYTHON_HEADERS}
-)
-
-target_link_libraries(PyHSPlasma HSPlasma ${PYTHON_LIBRARIES})
+Python3_add_library(PyHSPlasma MODULE
+                    ${DEBUG_SOURCES}        ${DEBUG_HEADERS}
+                    ${MATH_SOURCES}         ${MATH_HEADERS}
+                    ${PRP_ANIM_SOURCES}     ${PRP_ANIM_HEADERS}
+                    ${PRP_AUDIO_SOURCES}    ${PRP_AUDIO_HEADERS}
+                    ${PRP_AVATAR_SOURCES}   ${PRP_AVATAR_HEADERS}
+                    ${PRP_CAMERA_SOURCES}   ${PRP_CAMERA_HEADERS}
+                    ${PRP_COND_SOURCES}     ${PRP_COND_HEADERS}
+                    ${PRP_GEOM_SOURCES}     ${PRP_GEOM_HEADERS}
+                    ${PRP_GUI_SOURCES}      ${PRP_GUI_HEADERS}
+                    ${PRP_KEYOBJ_SOURCES}   ${PRP_KEYOBJ_HEADERS}
+                    ${PRP_LIGHT_SOURCES}    ${PRP_LIGHT_HEADERS}
+                    ${PRP_MSG_SOURCES}      ${PRP_MSG_HEADERS}
+                    ${PRP_MISC_SOURCES}     ${PRP_MISC_HEADERS}
+                    ${PRP_MOD_SOURCES}      ${PRP_MOD_HEADERS}
+                    ${PRP_OBJ_SOURCES}      ${PRP_OBJ_HEADERS}
+                    ${PRP_PARTICLE_SOURCES} ${PRP_PARTICLE_HEADERS}
+                    ${PRP_PHYS_SOURCES}     ${PRP_PHYS_HEADERS}
+                    ${PRP_REGION_SOURCES}   ${PRP_REGION_HEADERS}
+                    ${PRP_SURFACE_SOURCES}  ${PRP_SURFACE_HEADERS}
+                    ${PRP_SOURCES}          ${PRP_HEADERS}
+                    ${RESMGR_SOURCES}       ${RESMGR_HEADERS}
+                    ${SDL_SOURCES}          ${SDL_HEADERS}
+                    ${STREAM_SOURCES}       ${STREAM_HEADERS}
+                    ${SYS_SOURCES}          ${SYS_HEADERS}
+                    ${UTIL_SOURCES}         ${UTIL_HEADERS}
+                    ${VAULT_SOURCES}        ${VAULT_HEADERS}
+                    ${PYTHON_SOURCES}       ${PYTHON_HEADERS}
+)
+
+target_link_libraries(PyHSPlasma PUBLIC HSPlasma)
 set_target_properties(PyHSPlasma PROPERTIES PREFIX "")
 
 if(NOT WIN32)
-    set_target_properties(PyHSPlasma PROPERTIES
-                          SUFFIX ".so"
-    )
-
-    execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "import sys, distutils.sysconfig; sys.stdout.write(distutils.sysconfig.get_python_lib(prefix='${CMAKE_INSTALL_PREFIX}'))"
-                    OUTPUT_VARIABLE _PYTHON_LIB_DIR
+    execute_process(
+        COMMAND "${Python3_EXECUTABLE}" -c
+        "import sys, sysconfig; sys.stdout.write(sysconfig.get_path('purelib', vars={'base': '${CMAKE_INSTALL_PREFIX}'}))"
+        OUTPUT_VARIABLE _PYTHON_LIB_DIR
     )
     install(TARGETS PyHSPlasma
             DESTINATION ${_PYTHON_LIB_DIR}
     )
     install(FILES PyHSPlasma.pyi DESTINATION ${_PYTHON_LIB_DIR})
 else()
-    set_target_properties(PyHSPlasma PROPERTIES
-                          SUFFIX ".pyd"
-    )
-
     install(TARGETS PyHSPlasma
             RUNTIME DESTINATION bin
-            LIBRARY DESTINATION lib
+            LIBRARY DESTINATION bin
```

Comment:
CMake's FindPython3 (source of `Python3_add_library`) thinks Python extension modules are libraries. 🙄